Typescript migration

anish000kumar opened the pull request:
Added basic support for typescript.

Commit:
typescript first version

Commit:
updated helper

Commit:
added types for module and storeConfig

Commit:
typescript migration done

Huvik commented:
There are still missing types for `@connect` decorator for example. It would be really nice to have it so you have selector typed etc.

anish000kumar committed:
Update package.json

Commit:
removed package lock

anish000kumar committed:
Update readme.md

anish000kumar committed:
Update readme.md

anish000kumar committed:
Update readme.md

Commit:
fix dependecy mismatch issue

Commit:
upgraded version to 2.0.0 and updated docs

- Added JSdoc for each functions
- Simplified store implementation

anish000kumar committed:
Updated readme

anish000kumar committed:
Update readme.md

Commit:
built artifacts in dist folder

Commit:
Merge branch 'master' of https://github.com/anish000kumar/redux-box

Commit:
updated readme and build config

anish000kumar committed:
Update readme.md

Commit:
Integrated Travis and Coveralls for CI and coverage report

Commit:
Merge branch 'master' of https://github.com/anish000kumar/redux-box

Commit:
updated Readme

anish000kumar committed:
Update readme.md

anish000kumar committed:
Create FUNDING.yml

anish000kumar committed:
Update FUNDING.yml

anish000kumar committed:
Update FUNDING.yml

dependabot[bot] committed:
Bump semantic-release from 15.13.24 to 17.2.3

Bumps [semantic-release](https://github.com/semantic-release/semantic-release) from 15.13.24 to 17.2.3.
- [Release notes](https://github.com/semantic-release/semantic-release/releases)
- [Commits](https://github.com/semantic-release/semantic-release/compare/v15.13.24...v17.2.3)

Signed-off-by: dependabot[bot] <support@github.com>

anish000kumar committed:
Merge pull request #35 from anish000kumar/dependabot/npm_and_yarn/semantic-release-17.2.3

Bump semantic-release from 15.13.24 to 17.2.3

dependabot[bot] committed:
Bump immer from 3.2.0 to 8.0.1

Bumps [immer](https://github.com/immerjs/immer) from 3.2.0 to 8.0.1.
- [Release notes](https://github.com/immerjs/immer/releases)
- [Commits](https://github.com/immerjs/immer/compare/v3.2.0...v8.0.1)

Signed-off-by: dependabot[bot] <support@github.com>

anish000kumar committed:
Merge pull request #36 from anish000kumar/dependabot/npm_and_yarn/immer-8.0.1

Bump immer from 3.2.0 to 8.0.1

dependabot[bot] committed:
Bump terser from 4.2.1 to 4.8.1

Bumps [terser](https://github.com/terser/terser) from 4.2.1 to 4.8.1.
- [Release notes](https://github.com/terser/terser/releases)
- [Changelog](https://github.com/terser/terser/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terser/terser/commits)

---
updated-dependencies:
- dependency-name: terser
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump decode-uri-component from 0.2.0 to 0.2.2

Bumps [decode-uri-component](https://github.com/SamVerschueren/decode-uri-component) from 0.2.0 to 0.2.2.
- [Release notes](https://github.com/SamVerschueren/decode-uri-component/releases)
- [Commits](https://github.com/SamVerschueren/decode-uri-component/compare/v0.2.0...v0.2.2)

---
updated-dependencies:
- dependency-name: decode-uri-component
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump qs from 6.5.2 to 6.5.3

Bumps [qs](https://github.com/ljharb/qs) from 6.5.2 to 6.5.3.
- [Release notes](https://github.com/ljharb/qs/releases)
- [Changelog](https://github.com/ljharb/qs/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ljharb/qs/compare/v6.5.2...v6.5.3)

---
updated-dependencies:
- dependency-name: qs
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump ua-parser-js from 0.7.20 to 0.7.33

Bumps [ua-parser-js](https://github.com/faisalman/ua-parser-js) from 0.7.20 to 0.7.33.
- [Release notes](https://github.com/faisalman/ua-parser-js/releases)
- [Changelog](https://github.com/faisalman/ua-parser-js/blob/master/changelog.md)
- [Commits](https://github.com/faisalman/ua-parser-js/compare/0.7.20...0.7.33)

---
updated-dependencies:
- dependency-name: ua-parser-js
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

anish000kumar committed:
Merge pull request #46 from anish000kumar/dependabot/npm_and_yarn/ua-parser-js-0.7.33

Bump ua-parser-js from 0.7.20 to 0.7.33

anish000kumar committed:
Merge pull request #45 from anish000kumar/dependabot/npm_and_yarn/qs-6.5.3

Bump qs from 6.5.2 to 6.5.3

anish000kumar committed:
Merge pull request #44 from anish000kumar/dependabot/npm_and_yarn/decode-uri-component-0.2.2

Bump decode-uri-component from 0.2.0 to 0.2.2

dependabot[bot] committed:
Bump lodash from 4.17.15 to 4.17.21

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.15 to 4.17.21.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.15...4.17.21)

---
updated-dependencies:
- dependency-name: lodash
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

anish000kumar committed:
Merge pull request #43 from anish000kumar/dependabot/npm_and_yarn/terser-4.8.1

Bump terser from 4.2.1 to 4.8.1

dependabot[bot] committed:
Bump trim-off-newlines from 1.0.1 to 1.0.3

Bumps [trim-off-newlines](https://github.com/stevemao/trim-off-newlines) from 1.0.1 to 1.0.3.
- [Release notes](https://github.com/stevemao/trim-off-newlines/releases)
- [Commits](https://github.com/stevemao/trim-off-newlines/compare/v1.0.1...v1.0.3)

---
updated-dependencies:
- dependency-name: trim-off-newlines
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

anish000kumar committed:
Merge pull request #47 from anish000kumar/dependabot/npm_and_yarn/lodash-4.17.21

Bump lodash from 4.17.15 to 4.17.21

anish000kumar committed:
Merge pull request #48 from anish000kumar/dependabot/npm_and_yarn/trim-off-newlines-1.0.3

Bump trim-off-newlines from 1.0.1 to 1.0.3

molang163 committed:
Add async API example docs

anish000kumar committed:
Add PR test workflow for Node.js

This workflow runs tests on pull requests targeting the master branch using Node.js.

anish000kumar committed:
Merge pull request #52 from apples-kksk/add-async-api-example

Add advanced async API example

akumar52at committed:
Upgrade to latest versions

anish000kumar committed:
Merge pull request #54 from anish000kumar/upgrade-versions

Upgrade to latest versions

anish000kumar committed:
Add GitHub Actions workflow for NPM publishing

anish000kumar committed:
Update publish.yml

anish000kumar committed:
Change yarn install command to 'yarn'

anish000kumar committed:
Update publish.yml

anish000kumar committed:
coveralls upgraded and docs polish

anish000kumar committed:
more changes

anish000kumar committed:
Merge pull request #55 from anish000kumar/upgrade-versions

Upgrade versions

anish000kumar committed:
Enhance GitHub Actions workflow with permissions and build steps

Added permissions for GitHub Actions and updated build steps to include documentation generation.

anish000kumar committed:
add workflow

anish000kumar committed:
Merge pull request #56 from anish000kumar/workflo-update

add workflow

anish000kumar committed:
Fix typos in workflow files

anish000kumar committed:
Merge pull request #57 from anish000kumar/workflo-update

Fix typos in workflow files

anish000kumar committed:
Update docs

anish000kumar committed:
Update docs

anish000kumar committed:
Merge pull request #58 from anish000kumar/workflo-update

Workflo update

anish000kumar committed:
Add graphql example and update testing guide

anish000kumar committed:
Merge pull request #59 from anish000kumar/gql-example

Add graphql example and update testing guide

anish000kumar committed:
Merge branch 'master' into typescript-migration

anish000kumar committed:
Migrated to TS

anish000kumar committed:
remove obsolete config
